### PR TITLE
Alternative error handling for `fsm_print_c`

### PR DESCRIPTION
**Pull request by Caslyn**
This PR returns `-1` instead of asserting or returning `EOF` in error scenarios.  

The reasons for these changes are: 
- Make error handling more flexible for the caller of the fsm function; for example, handling a `-1` return value instead of dying on an `assert` 
- Making `-1` a more explicit error value instead of `EOF`; reducing the need for the context to handle or understand an `EOF` value.
- Eliminate the need to include libraries for `EOF`/`assert` in the generated code

Without these changes, a context that wants to handle `NULL` (or empty string) input differently would need to maintain that handling separately (and have it executed before the libfsm-generated code is called); additionally it would be confined to using `.fragment = true` if it wants this error handling in the same function as the state machine.

### CODE REVIEW

File: src/libfsm/print/c.c [hunk 1]
```diff
@@ -423,7 +421,9 @@ fsm_print_c(FILE *f, const struct fsm *fsm)
 		fprintf(f, "{\n");
 		fprintf(f, "\tint c;\n");
 		fprintf(f, "\n");
-		fprintf(f, "\tassert(fsm_getc != NULL);\n");
+		fprintf(f, "\tif (fsm_getc == NULL) {\n");
```
Comment (silentbicycle) [inline src/libfsm/print/c.c]: Is fsm_getc being NULL here potentially a recoverable error?

File: src/libfsm/print/c.c [hunk 2]
```diff
@@ -432,7 +432,9 @@ fsm_print_c(FILE *f, const struct fsm *fsm)
 		fprintf(f, "{\n");
 		fprintf(f, "\tconst char *p;\n");
 		fprintf(f, "\n");
-		fprintf(f, "\tassert(s != NULL);\n");
+		fprintf(f, "\tif (s == NULL) {\n");
+		fprintf(f, "\t\treturn -1;\n");
```
Comment (silentbicycle) [inline src/libfsm/print/c.c]: Is this recoverable? If so, should it be distinct from the -1 `EOF` result?

File: src/libfsm/print/c.c (hunk 1 above)
Comment (Caslyn) [reply src/libfsm/print/c.c]: Undefined behavior it is! dc52d7c

File: src/libfsm/print/c.c (hunk 2 above)
Comment (Caslyn) [reply src/libfsm/print/c.c]: Undefined behavior it is! dc52d7c

**Pull request merged**